src/wechaty/user/friendship.py

```diff
@@ -59,18 +59,12 @@ def __init__(self, friendship_id: str):
         """
         initialization constructor for friendship
         """
+        super(Friendship, self).__init__()
+
         self.friendship_id = friendship_id
-        self._payload: Optional[FriendshipPayload] = None
 
         log.info('Friendship constructor %s', friendship_id)
 
-        # if self.__class__ is Friendship:
-        #     raise Exception(
-        #         'Friendship class can not be instanciated directly!')
-        if self.puppet is None:
-            raise Exception(
-                'Friendship class can not be instanciated without a puppet!')
-
```

Comment:
Also, the above code intent to restrict the user to create any acessory classes only when the bot/puppet are all instantiated, or else raises an Error, but this piece of code cannot pass the `accessory_test`, specifically the `get_user_class() ` function.
So we may need to look through about `accessory_test` again after this refactoration.